### PR TITLE
Add case-insensitive configuration for keywords parsing

### PR DESCRIPTION
**Pull request by Pluralia**
Closes [#315](https://github.com/langium/langium/issues/315).
Adds a flag `caseInsensitive` in `LanguageMetaData`.

### CODE REVIEW

File: packages/langium/src/parser/token-builder.ts
Comment (spoenemann) [inline packages/langium/src/parser/token-builder.ts:15]: I'd propose to wrap this in an object:
```suggestion
    buildTokens(grammar: Grammar, options?: { caseInsensitive?: boolean }): TokenType[];
```
This is better extensible and avoids a Boolean function parameter, which I regard as bad API in general.

File: packages/langium/src/utils/regex-util.ts
Comment (spoenemann) [inline packages/langium/src/utils/regex-util.ts:119]: Please use the `map` function rather than a `for` loop with `push`.

File: examples/arithmetics/example/example.calc
Comment (spoenemann) [inline examples/arithmetics/example/example.calc:1]: Please don't forget to revert the example before merging.

**Review by spoenemann — APPROVED** [review]
Good work! A few detail remarks below, feel free to merge after addressing them.

File: examples/arithmetics/example/example.calc
Comment (Pluralia) [reply examples/arithmetics/example/example.calc:1]: I'd propose to leave it to demonstrate the ability to make keywords case-insensitive.

**Pull request merged**
